device.go

```diff
@@ -229,37 +230,182 @@ func virtioBlkDeviceHandler(_ context.Context, device pb.Device, spec *pb.Spec,
 		device.VmPath = devPath
 	}
 
-	return updateSpecDeviceList(device, spec)
+	return updateSpecDevice(spec, devIdx, device.ContainerPath, device.VmPath, device.ContainerPath)
 }
 
 // device.Id should be the SCSI address of the disk in the format "scsiID:lunID"
-func virtioSCSIDeviceHandler(ctx context.Context, device pb.Device, spec *pb.Spec, s *sandbox) error {
+func virtioSCSIDeviceHandler(ctx context.Context, device pb.Device, spec *pb.Spec, s *sandbox, devIdx devIndex) error {
 	// Retrieve the device path from SCSI address.
 	devPath, err := getSCSIDevPath(s, device.Id)
 	if err != nil {
 		return err
 	}
 	device.VmPath = devPath
 
-	return updateSpecDeviceList(device, spec)
+	return updateSpecDevice(spec, devIdx, device.ContainerPath, device.VmPath, device.ContainerPath)
 }
 
-func nvdimmDeviceHandler(_ context.Context, device pb.Device, spec *pb.Spec, s *sandbox) error {
-	return updateSpecDeviceList(device, spec)
+func nvdimmDeviceHandler(_ context.Context, device pb.Device, spec *pb.Spec, s *sandbox, devIdx devIndex) error {
+	return updateSpecDevice(spec, devIdx, device.ContainerPath, device.VmPath, device.ContainerPath)
 }
 
-// updateSpecDeviceList takes a device description provided by the caller,
-// trying to find it on the guest. Once this device has been identified, the
-// "real" information that can be read from inside the VM is used to update
-// the same device in the list of devices provided through the OCI spec.
-// This is needed to update information about minor/major numbers that cannot
-// be predicted from the caller.
-func updateSpecDeviceList(device pb.Device, spec *pb.Spec) error {
-	// If no ContainerPath is provided, we won't be able to match and
-	// update the device in the OCI spec device list. This is an error.
-	if device.ContainerPath == "" {
+// Take the guest device with the given DDDD:BB:SS.F PCI address,
```

Comment:
Should the comment refer to "DDDD:BB:DD.F" ? (Domain:Bus:Dev.Func )

Reply:
I guess `S` stands for slot? `D` is already taken by "Domain", so having it twice would be ambiguous.